### ISSUE TITLE
unable to change submission status 

### ISSUE THREAD

**Issue by manjitapandey** [hotosm/fmtm#1581]
**Describe the bug**
I am unable to change submission status from pending to any other. No API hit is there and the pop-up behaves like static. 

**To Reproduce**

1. Go to 'submission table page'
2. Click on 'tick icon on right of the table '
3. Change submission status from pending to any other
5. See error

**Expected behavior**
The submission status should be changed to respective status. 

**Screenshots**
https://vimeo.com/960121064/2e5629ee22?share=copy
![image](https://github.com/hotosm/fmtm/assets/97273021/0833a475-a6ab-42ba-a12c-a399d72ae6fa)

**Comment by NSUWAL123** [comment]
Previously the task_id which is essential for updating the review state of a submission was retrieved from the all object
![image_720](https://github.com/hotosm/fmtm/assets/81785002/cd7a1ade-d392-4a37-99cd-91ea369aabd9)


but now it seems to come under essential key
![image_480](https://github.com/hotosm/fmtm/assets/81785002/36f906a8-6baa-4422-bd38-a19c7b545c5f)

for quick fix should I just update the key name to `essential` or should finalize in which key the building `line string` & `taskId` would come

**Comment by spwoodcock** [comment]
Answered this via slack message.

We are updating to be under the essential key and this likely will not change in the future.

However the more versatile fix is to use flatjson for the submission data on the API

**Comment by spwoodcock** [comment]
Slack chat dump:

Sam Woodcock
  11 hours ago
If we move the fields outside of the essential group in the XLSForm, it should fix the issue (as it essentially just reverts to how it was before) (edited) 

Sam Woodcock
  [11 hours ago](https://hotosm.slack.com/archives/C0738H93AKT/p1718744731609019?thread_ts=1718610733.238329&cid=C0738H93AKT)
The XLSForms are so finnicky and prone to breakage - the form for the project you were testing should be downloaded, modified, and reuploaded for testing

Sam Woodcock
  [11 hours ago](https://hotosm.slack.com/archives/C0738H93AKT/p1718744778974099?thread_ts=1718610733.238329&cid=C0738H93AKT)
The digitisation_correct field should trigger an update to the status field. I just tested removing the essential group and verificiation group & it does work updating the status value in the submission

Sam Woodcock
  [11 hours ago](https://hotosm.slack.com/archives/C0738H93AKT/p1718745028570779?thread_ts=1718610733.238329&cid=C0738H93AKT)
The entity status is still not updated, but I think this is probably because the link between the XLSForm field andthe entity field is made when the Entity is created. As the entity was generated when the status field was still nested under essetial , perhaps the linkage was not made correctly

Sam Woodcock
  [11 hours ago](https://hotosm.slack.com/archives/C0738H93AKT/p1718745148754809?thread_ts=1718610733.238329&cid=C0738H93AKT)
In summary we need to update the XLSForm in osm-fieldwork to remove the groups essential and verification, plus also ideally remove the form_category field too.

**Comment by spwoodcock** [comment]
I'll take a look this morning! Will update if I solve it 👍 
If I run out of time, I will re-assign if that's ok 😄

**Comment by NSUWAL123** [comment]
Sure

**Comment by spwoodcock** [comment]
@NSUWAL123 I think I have worked out the issues.

I don't think we should rely on the form submission for the `task_id`.
In SubmissionsTable.tsx, can we instead get the taskId from the URL: `searchParams.get('task_id')`?
(this is done elsewhere in the file too 👍)

**Comment by spwoodcock** [comment]
I updated the file as described above. Hopefully it fixes the issue: https://github.com/hotosm/fmtm/commit/fa3b235c9a48a0c1e0df9160bcf901a9515149dd

**Comment by spwoodcock** [comment]
Hold on, I thought this and #1576 were basically the same issue, but they are not.
Just to clarify, this issue is not about **submission** status, it's about **review status**, as in the video 👍

**Comment by spwoodcock** [comment]
@NSUWAL123 can you foresee the change in this commit creating any issues?
https://github.com/hotosm/fmtm/commit/fa3b235c9a48a0c1e0df9160bcf901a9515149dd

If so, we can revert to use the task_id from the submission.
The key would now be `taskId: row?.task_id` instead of `taskId: row?.all?.task_id` (the `all` group from removed from the first version of the XLSForms).

**Comment by NSUWAL123** [comment]
@spwoodcock, But why is it `searchParams.get('task_id")` ?

**Comment by spwoodcock** [comment]
Damn I stupidly assumed this was URL search params, but it's search params for the table right?

In that case of course this makes no sense 🤦 

I'll revert it!

**Comment by spwoodcock** [comment]
Updated https://github.com/hotosm/fmtm/commit/fbc2b376e02ee0ac2b03e2ee8f9e259d3d33b7d7

**Comment by manjitapandey** [comment]
@spwoodcock the issue is still the same.

**Comment by spwoodcock** [comment]
Sorry I think we discussed two issues in this thread.

The one for displaying the filtered submissions in the submissions table is fixed.

The original issue related to review state I haven't looked at. I think that can go in the next release - what do you think?

**Comment by manjitapandey** [comment]
I don't think we should wait for the next release to fix this review status update, Since it can be important for validator user. The major task for validator team would be validate the submission but due to this issue, the workflow for them seems to be breaking.

**Comment by spwoodcock** [comment]
Fair enough 👍

I haven't considered validators important until now, as we don't even have the role implemented.

My main priority has always been to get users mapping & allow the manager to extract the end data.

**Comment by NSUWAL123** [comment]
@spwoodcock, I will fix that review state issue quickly. the issue was mainly due to task_id null, since task_id is back, I will update the fields accordingly and make a PR ASAP
